editor/src/components/navigator/left-pane/pages-pane.tsx

```diff
@@ -0,0 +1,234 @@
+  }, [navigationControls, activeRemixScene, props.resolvedPath])
+
+  const resolvedRouteSegments = props.resolvedPath.split('/')
+  const templateRouteSegments = props.routePath.split('/')
+
+  const lastResolvedSegment = resolvedRouteSegments[resolvedRouteSegments.length - 1]
+  const lastTemplateSegment = templateRouteSegments[templateRouteSegments.length - 1]
+
+  const indentation = resolvedRouteSegments.length - 2
+
+  const isDynamicPathSegment = lastTemplateSegment.startsWith(':')
+
+  return (
+    <FlexRow
+      style={{
+        flexShrink: 0,
+        color: colorTheme.neutralForeground.value,
+        backgroundColor: props.active ? colorTheme.subtleBackground.value : 'transparent',
+        marginLeft: 8,
+        marginRight: 8,
+        paddingLeft: 3 + indentation * 15,
+        paddingTop: 3,
+        paddingBottom: 3,
+        height: UtopiaTheme.layout.rowHeight.smaller,
+        alignItems: 'center',
+        borderRadius: 2,
+        position: 'relative',
+      }}
+      onClick={props.matchesRealRoute ? onClick : NO_OP}
+    >
+      <ExpandableIndicator
+        key='expandable-indicator'
+        visible={true}
+        collapsed={false}
+        selected={false}
+        style={{ transform: 'scale(0.6)', opacity: 'var(--paneHoverOpacity)' }}
+        iconColor={'main'}
+      />
+      <Icn
+        style={{
+          marginRight: 0,
+          transform: 'scale(.85)',
+        }}
+        category='filetype'
+        color={'main'}
+        type={props.matchesRealRoute ? 'other' : 'folder-open'}
+        width={18}
+        height={18}
+      />
+      {/* TODO if we want renaming, cannibalize it from FileBrowserItem */}
+      <span
+        style={{
+          marginLeft: 6,
+          display: 'inline-block',
+          whiteSpace: 'nowrap',
+          overflow: 'hidden',
+          textOverflow: 'ellipsis',
+          flexGrow: 1,
+        }}
+      >
+        {lastResolvedSegment === '' ? RemixIndexPathLabel : '/' + lastResolvedSegment}
+      </span>
+      <span
+        style={{
+          flexShrink: 0,
+          display: !isDynamicPathSegment ? 'none' : props.active ? 'inline-block' : undefined, // I'm sorry
+          color: colorTheme.subduedForeground.value,
+          marginLeft: 6,
+          whiteSpace: 'nowrap',
+          overflow: 'hidden',
+          textOverflow: 'ellipsis',
+          paddingRight: 8,
+          paddingLeft: 2,
+        }}
+        css={{
+          display: 'none',
+          '*:hover > &': {
+            display: 'inline-block',
+          },
+        }}
+      >
+        {lastTemplateSegment}
+      </span>
+    </FlexRow>
+  )
+})
+
+function fillInGapsInRoute(routes: Array<string>): Array<string> {
+  // if we find a route /collections, and a route /collections/hats/beanies, we should create an entry for /collections/hats, so there are no gaps in the tree structure
+  const result: Array<string> = []
```

Comment:
Our convention is to use `let` when we mutate the variable later (even though I don't like that convention :) )